packages/react-components/react-popover/src/components/Popover/usePopover.ts

```diff
@@ -57,7 +57,33 @@ export const usePopover_unstable = (props: PopoverProps): PopoverState => {
     popoverSurface = children[0];
   }
 
-  const [open, setOpen] = useOpenState(initialState);
+  const [open, setOpenState] = useOpenState(initialState);
+
+  const setOpenTimeoutRef = React.useRef(0);
+
+  const setOpen = useEventCallback((e: OpenPopoverEvents, shouldOpen: boolean) => {
+    clearTimeout(setOpenTimeoutRef.current);
+
+    if (e.type === 'mouseleave') {
+      // FIXME leaking Node timeout type
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      // @ts-ignore
+      setOpenTimeoutRef.current = setTimeout(() => {
+        setOpen(e, shouldOpen);
+      }, props.mouseLeaveDelay ?? 500);
```

Comment:
React <17 events are still pooled consider something simiar to this:
https://github.com/microsoft/fluentui/blob/dce35e473cb5940cd7b0da2a039ade5bfd29f9bd/packages/react-components/react-menu/src/components/Menu/useMenu.tsx#L157-L160